student.py

```diff
@@ -0,0 +1,99 @@
+import numpy as np
+import pandas as pd
+from pulp import *
+
+class Student:
+    '''Stores each student's info such as schedule, faculty interests, names, etc and provides method for optimizing meeting schedules'''
+    def __init__(self, legal_name, pref_name, faculty_interest = []):
+        self.legal_name = legal_name
+        self.pref_name = pref_name
+        self.faculty_interest = faculty_interest
+        self.schedule = None
+
+    def optimize_schedule(self, prof_list, prof_availability, sessions_dict, num_sessions):
+        # Defining optimization goal of maximizing the objective.
+        prob = LpProblem("Professor_Match", LpMaximize)
+
+        # Create filtered faculty matrix for student's interests
+        student_no_interest = np.setdiff1d(prof_list, self.faculty_interest)
+        student_no_interest_indices = np.where(np.isin(prof_list, student_no_interest))
+        interested_prof_avail = np.delete(prof_availability, student_no_interest_indices, 1)
+        interested_prof_list = np.delete(prof_list, student_no_interest_indices)
+
+        # Define range of meeting slots and professors
+        meeting_slots = range(num_sessions)
+        profs = range(interested_prof_avail.shape[1])
```

Comment:
probably better to name this something like `prof_count`, and just use `range(prof_count)` instead of trying to reuse the same range variable. `profs` sounds like this is a list of professors.

Reply:
kinda the same comment about `meeting_slots`